### DISCUSSION TITLE
problem with delete cookies

### DISCUSSION THREAD

**Discussion by alirezamehrabi** [vercel/next.js#59324] [Help] [answered]
### Summary

hi, in nextjs 14 i have code below for deleting all cookies
```

'use server'
import { cookies } from 'next/headers'

const DeleteCookie = (x) => {
        return cookies().getAll().forEach((cookie) => {
          cookies().delete(cookie.name);
        });
      };
export default DeleteCookie
```

I import that in a serverside component
`import DeleteAllCookies from "../Components/DeleteAllCookies";`

and pass it to a client-side component by a prop
`<SideBar DeleteAllCookies={DeleteAllCookies}/>`

I invoke it in a button
`<button onClick={()=>{DeleteAllCookies();router.push("/")}}>Exit</button>`

it works in localhost but after deploying website it has error below

Uncaught (in promise) Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.

**Comment by alexcraviotto** [comment]
Did you fix it?

**Reply by JoaoVitorSD** [reply]
Same issue here. Did you find the error? I'm stuck on this.

**Reply by alirezamehrabi** [reply, marked as answer]
Yes, deleting should be serverSide.
for exapmple for sign out you can do this:

api / signOutApi / route.ts

```
import { cookies } from 'next/headers'
import { NextResponse } from 'next/server';

export const GET = () => {
  cookies().getAll().forEach((cookie) => {
    cookies().delete(cookie.name);
  });
  return NextResponse.json({ status: 200, message: "You Signed Out Succesfully" })
}
```

now just call the api everywhere you want to signout and delete cookies

**Reply by JoaoVitorSD** [reply]
I only had this problema running in kubernetes pod. My solution was change to lib _nookies_.

https://www.npmjs.com/package/nookies
